nav2_costmap_2d/src/costmap_2d_publisher.cpp

```diff
@@ -220,6 +243,9 @@ void Costmap2DPublisher::publishCostmap()
       }
       costmap_update_pub_->publish(std::move(update));
     }
+    if (costmap_raw_update_pub_->get_subscription_count() > 0) {
+      costmap_raw_update_pub_->publish(createRawCostmapUpdateMsg());
```

Comment:
In this case, I believe that using `std::move() `explicitly might cause an unnecessary call to the move constructor of `unique_ptr`. 
I have run a toy example which showed that due to RVO/NRVO optimization, no additional call to the constructor was made. However, when calling std::move() I triggered the move constructor.
When I turned off the optimization I also had a call to move constructor. 

So to sum up, we are ending up with the move constructor call in the worst-case scenario, but when calling move, it is more likely, because it seems like it blocks RVO.